libs/sdk-core/src/breez_services.rs

```diff
@@ -905,11 +907,51 @@ impl BreezServices {
         Ok(())
     }
 
+    fn persist_new_bolt11_payment_attempt(
+        &self,
+        invoice: &LNInvoice,
+        amount_msat: u64,
+    ) -> Result<(), SendPaymentError> {
+        self.persister.insert_or_update_payments(&[Payment {
+            id: invoice.payment_hash.clone(),
+            payment_type: PaymentType::Sent,
+            payment_time: SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs() as i64,
+            amount_msat: 0,
+            fee_msat: 0,
+            status: PaymentStatus::Pending,
+            description: invoice.description.clone(),
+            details: PaymentDetails::Ln {
+                data: LnPaymentDetails {
+                    payment_hash: invoice.payment_hash.clone(),
+                    label: String::new(),
+                    destination_pubkey: invoice.payee_pubkey.clone(),
+                    payment_preimage: String::new(),
+                    keysend: false,
+                    bolt11: invoice.bolt11.clone(),
+                    lnurl_success_action: None,
+                    ln_address: None,
+                    lnurl_metadata: None,
+                    lnurl_withdraw_endpoint: None,
+                    swap_info: None,
+                },
+            },
+        }])?;
+
+        self.persister.insert_payment_external_info(
```

Comment:
Why do we need to add the amount to the sync db? Is there a reason to sync pending payments?

Reply:
Because the `attempted_amount_msat` is also used for failed open invoices to display the actual amount that was attempted to be paid.
And I assume, although not a 100% sure, that other signers would be able to sync the pending payment from Greenlight as well and thus need the data to display the accurrate amount.

Reply:
But we already handled failed opened invoices. Why persist for every invoice? For success payments the amount is retrieved correctly.
I see disadvantage for this way of sync where the synced data become larger without any real advantage.

Reply:
I see your point, but if we don't sync the attempted amount we couldn't display the actual intended amount for pending payments on all remote signers, as `amount_msat` is only the [amount delivered to destination](https://docs.corelightning.org/reference/lightning-listsendpays#return-value) (we could display it on the sending signer if we would just store it locally), which seems problematic to me as this would result it inconsistencies.
The other option I see would be to clean up the synced entry after payment success.

I would like to get your input on this

Reply:
> I see your point, but if we don't sync the attempted amount we couldn't display the actual intended amount for pending payments on all remote signers, as `amount_msat` is only the [amount delivered to destination](https://docs.corelightning.org/reference/lightning-listsendpays#return-value) (we could display it on the sending signer if we would just store it locally), which seems problematic to me as this would result it inconsistencies. The other option I see would be to clean up the synced entry after payment success.
> 
> I would like to get your input on this

The signer is synced with the channel state from greenlight not using our own sync db mechanism to it is irrelevant to the signer sync.
The sdk itself using the sync db to synchronize but it doesn't do that in realtime and we don't have a way to notify remote sdk instances to sync so I don't see how adding it to the sync db contributes here.
In addition immediately after the send_payment is called remote sdk that will sync (actually call list_pays behind the scene) will receive the pending payment as pending so that will should your concern as well.

Reply:
Do I understand this correctly; Other signers might get the pending payment (`list_pays`) before they sync the remote DB?

Reply:
> Do I understand this correctly; Other signers might get the pending payment (`list_pays`) before they sync the remote DB?

In short Yes, exactly.
Let me elaborate a bit on the way we sync data across sdk instances which should make it even more clear:
there are two ways we synchronize:
1. Sync remote data that exists on greenlight (payments, invoices, channels, etc...) this data comes from greenlight and is fetched by simple calling greenlight gRpc API. All of it is encapsulated inside the `sync` method.
2. Sync remote data that doesn't exist in greenlight and only the sdk "knows" about it (lnurl metadata, swaps, etc...) for this kind of data we use the sync.db and whenever such data changes locally the sdk ensures it is uploaded to the greenlight storage (using internal `backup` API) so other sdk instances can pull it.

The pending payments are related to the first kind of sync (greenlight data) so in theory we don't need to do anything about it as if one sdk instance pays, other instances will receive it as soon as they call `sync`.
Your requirement was a local requirement as far as I understand, to be able to see the pending payment **immediately** after calling `send_payment`.  For that purpose only adding locally is needed and no need to add it to the sync.db as well.
Hope that makes sense.